lib/validation/bundlers/constraintsVB.ts

```diff
@@ -1,201 +1,158 @@
+          tag: this.getTag(),
+          target: constraints,
+          predicate: (constraints: Constraints) => ConstraintsVB.shouldBeKnownOption(constraints.is_holder),
+          message: this.subjectIsIssuerShouldBeKnownValueMsg,
+        },
+        {
+          tag: this.getTag(),
+          target: constraints,
+          predicate: (constraints: Constraints) =>
+            this.fieldIdInSubjectMustCorrespondToFieldId(constraints, constraints.is_holder),
+          message: this.fieldIdMustCorrespondToFieldIdMsg,
+        },
+        {
+          tag: this.getTag(),
+          target: constraints,
+          predicate: (constraints: Constraints) =>
+            this.fieldIdInSubjectMustCorrespondToFieldId(constraints, constraints.same_subject),
+          message: this.fieldIdMustCorrespondToFieldIdMsg,
+        },
+        ...this.getSubjectsValidations(constraints?.is_holder),
+        ...this.getSubjectsValidations(constraints?.same_subject),
+        ...this.getFieldsValidations(constraints),
+      ];
+    }
+    return validations;
   }
 
-  private getFieldsValidations(constraints: Constraints): Validation<any>[] {
-    if (constraints && constraints.fields) {
+  private getFieldsValidations(constraints: Constraints): Validation<Field>[] {
+    if (constraints?.fields?.length) {
       return new FieldsVB(this.getTag()).getValidations(constraints.fields);
     }
     return [];
   }
 
-  private static disclosureLimitShouldHaveKnownValue(limit_disclosure: Optionality): boolean {
-    return (
-      limit_disclosure == null ||
-      limit_disclosure === Optionality.Preferred ||
-      limit_disclosure === Optionality.Required
-    );
+  private static disclosureLimitShouldHaveKnownValue(limit_disclosure?: Optionality): boolean {
+    return !limit_disclosure || limit_disclosure === Optionality.Preferred || limit_disclosure === Optionality.Required;
   }
 
-  private static statusShouldHaveKnownValue(statuses: Statuses): boolean {
+  private static statusShouldHaveKnownValue(statuses: Statuses | undefined): boolean {
     return statuses == null || statuses.active != null || statuses.revoked != null || statuses.suspended != null;
   }
 
-  private static statusDirectiveShouldHaveKnownValue(): ValidationPredicate<Statuses> {
-    return (statuses: Statuses): boolean =>
-      this.pdStatusShouldBeKnown(statuses?.active) &&
-      this.pdStatusShouldBeKnown(statuses?.revoked) &&
-      this.pdStatusShouldBeKnown(statuses?.suspended);
+  private static statusDirectiveShouldHaveKnownValue(): ValidationPredicate<Constraints> {
+    return (constraints: Constraints): boolean =>
+      this.pdStatusShouldBeKnown(constraints?.statuses?.active) &&
+      this.pdStatusShouldBeKnown(constraints?.statuses?.revoked) &&
+      this.pdStatusShouldBeKnown(constraints?.statuses?.suspended);
   }
 
   private static pdStatusShouldBeKnown(pdStatus: PdStatus | undefined): boolean {
     return (
-      !pdStatus ||
+      pdStatus == null ||
```

Comment:
What about undefined, which the previous code would catch

Reply:
Agreed!